hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -152,6 +154,24 @@ public static KeyGenerator createKeyGenerator(TypedProperties props) throws IOEx
     }
   }
 
+  /**
+   * Create a UserDefinedBulkInsertPartitioner class via reflection,
+   * <br>
+   * if the class name of UserDefinedBulkInsertPartitioner is configured through the HoodieWriteConfig.
+   * @see HoodieWriteConfig#getUserDefinedBulkInsertPartitionerClass()
+   */
+  private static Option<UserDefinedBulkInsertPartitioner> createUserDefinedBulkInsertPartitioner(HoodieWriteConfig config)
+          throws IOException {
+    String bulkInsertPartitionerClass = config.getUserDefinedBulkInsertPartitionerClass();
+    try {
+      return bulkInsertPartitionerClass == null || bulkInsertPartitionerClass.isEmpty()
+              ? Option.empty() :
+              Option.of((UserDefinedBulkInsertPartitioner) ReflectionUtils.loadClass(bulkInsertPartitionerClass));
+    } catch (Throwable e) {
+      throw new IOException("Could not create UserDefinedBulkInsertPartitioner class " + bulkInsertPartitionerClass, e);
```

Comment:
throw a HoodieException?  this is not an IOException per se?

Reply:
While I was reading the code of [DataSourceUtils](https://github.com/apache/incubator-hudi/blob/master/hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java#L58), I had the impression that it tries to throw IOException as a configuration issue of user's while Hoodie*Exception would means as more toward some sort of Hudi internal error.

For example, failed to load of ```hoodie.datasource.write.keygenerator.class``` class throws as IOException from [createKeyGenerator](https://github.com/apache/incubator-hudi/blob/master/hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java#L151), and ```hoodie.datasource.write.payload.class``` too from [createPayload](https://github.com/apache/incubator-hudi/blob/master/hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java#L164), so question is when a user mis-configure as class name, should it be considered as "hoodie" error? or general configuration error as user mistake?
I hope there is specific user configuration exception separately, but I wasn't sure the intention of IOException from other configurations, so I followed existing ways.

Please let me know if you think this should be HoodieException, I can update it.

Reply:
correct.. a cleanup of these exceptions would be a good follow up.. For now, let's try fixing in baby steps, by having this throw a generic HoodieException (both are uncaught for the user anwyay).

Reply:
Ok, I will change it.